src/test/java/org/jsoup/helper/HttpConnectionTest.java

```diff
@@ -160,6 +161,13 @@
         assertEquals("http://example.com", con.request().url().toExternalForm());
     }
 
+    @Test public void connectWithIdnUrl() throws MalformedURLException {
+        URL url = new URL("https://тест.рф");
+        Connection con = HttpConnection.connect(url, true);
```

Comment:
Need to make sure we have test for redirects to IDNs, and for finding IDN in HTML and following.

Reply:
Can't write tests. Have lack of knowledge.